app/src/main/java/com/sdpteam/connectout/event/nearbyEvents/list/EventsAdapter.java

```diff
@@ -0,0 +1,78 @@
+package com.sdpteam.connectout.event.nearbyEvents.list;
+
+import java.util.List;
+
+import com.sdpteam.connectout.R;
+import com.sdpteam.connectout.event.Event;
+import com.sdpteam.connectout.profile.ProfileActivity;
+
+import android.content.Context;
+import android.content.Intent;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ArrayAdapter;
+import android.widget.ImageButton;
+import android.widget.ImageView;
+import android.widget.TextView;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+/**
+ * It is responsible for creating and managing the views for a list of events.
+ * The class holds the context, resource, and a list of events, which are passed in through the constructor.
+ */
+public class EventsAdapter extends ArrayAdapter<Event> {
+
+    private final Context context;
```

Comment:
Don't need to keep the context here, you can access it from the superclass with getContext() ;)